dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -122,175 +133,48 @@ if (process.env.LC_ALL) {
-            height: windowState.height,
-            minWidth: 200,
-            minHeight: 120,
-            x: windowState.x,
-            y: windowState.y,
-            isMaximized: windowState.isMaximized
-        };
+// Set the electron version for both the dev and the production mode. (https://github.com/theia-ide/theia/issues/3254)
+// Otherwise, the forked backend processes will not know that they're serving the electron frontend.
+const { versions } = process;
+// @ts-ignore
+if (versions && typeof versions.electron !== 'undefined') {
+    // @ts-ignore
+    process.env.THEIA_ELECTRON_VERSION = versions.electron;
+}
 
-        // Always hide the window, we will show the window when it is ready to be shown in any case.
-        const newWindow = new BrowserWindow(windowOptions);
-        if (windowOptions.isMaximized) {
-            newWindow.maximize();
-        }
-        newWindow.on('ready-to-show', () => newWindow.show());
+const mainPath = resolve(__dirname, '..', 'backend', 'main');
+const indexHtmlPath = resolve(__dirname, '../../lib/index.html');
+const applicationName = \`${this.pck.props.frontend.config.applicationName}\`;
 
-        // Prevent calls to "window.open" from opening an ElectronBrowser window,
-        // and rather open in the OS default web browser.
-        newWindow.webContents.on('new-window', (event, url) => {
-            event.preventDefault();
-            shell.openExternal(url);
-        });
+const container = new Container();
+container.bind('theia-application-name').toConstantValue(applicationName);
+container.bind('theia-backend-main-path').toConstantValue(resolve(mainPath));
+container.bind('theia-index-html-path').toConstantValue(indexHtmlPath);
+container.load(electronMainApplicationModule);
 
-        // Save the window geometry state on every change
-        const saveWindowState = () => {
-            try {
-                let bounds;
-                if (newWindow.isMaximized()) {
-                    bounds = electronStore.get(WINDOW_STATE, {});
-                } else {
-                    bounds = newWindow.getBounds();
-                }
-                electronStore.set(WINDOW_STATE, {
-                    isMaximized: newWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                });
-            } catch (e) {
-                console.error("Error while saving window state.", e);
-            }
-        };
-        let delayedSaveTimeout;
-        const saveWindowStateDelayed = () => {
-            if (delayedSaveTimeout) {
-                clearTimeout(delayedSaveTimeout);
-            }
-            delayedSaveTimeout = setTimeout(saveWindowState, 1000);
-        };
-        newWindow.on('close', saveWindowState);
-        newWindow.on('resize', saveWindowStateDelayed);
-        newWindow.on('move', saveWindowStateDelayed);
+function load(raw) {
```

Comment:
It is only because for now, no packages are contribution any `electron-main` specific containers/contributions. I used the same pattern as for the frontend and backend application.